### PR TITLE
DXF DIMSTYLE, LEADER and MULTILEADER

### PR DESCRIPTION
**Pull request by atlight**
https://trac.osgeo.org/gdal/ticket/7111

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/dxf/ogr_autocad_services.cpp [hunk 1]
```diff
@@ -421,6 +421,74 @@ const unsigned char *ACGetColorTable()
...
 /************************************************************************/
+/*                       ACGetKnownDimStyleCodes()                      */
+/*                                                                      */
+/*      Gets a list of the DIMSTYLE codes that we care about. Array     */
+/*      terminates with a zero value.                                   */
+/************************************************************************/
+
+int* ACGetKnownDimStyleCodes()
+{
+	static int aiKnownCodes[] = {
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/dxf/ogr_autocad_services.cpp]: static const int

File: gdal/ogr/ogrsf_frmts/dxf/ogr_autocad_services.cpp [hunk 2]
```diff
@@ -421,6 +421,74 @@ const unsigned char *ACGetColorTable()
 }
 
 /************************************************************************/
+/*                       ACGetKnownDimStyleCodes()                      */
+/*                                                                      */
+/*      Gets a list of the DIMSTYLE codes that we care about. Array     */
+/*      terminates with a zero value.                                   */
+/************************************************************************/
+
+int* ACGetKnownDimStyleCodes()
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/dxf/ogr_autocad_services.cpp]: const int*

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxf_leader.cpp [hunk 3]
```diff
@@ -0,0 +1,646 @@
+/******************************************************************************
+ *
+ * Project:  DXF Translator
+ * Purpose:  Implements translation support for LEADER and MULTILEADER
+ *           elements as a part of the OGRDXFLayer class.
+ * Author:   Alan Thomas, alant@outlook.com.au
+ *
+ ******************************************************************************
+ * Copyright (c) 2009, Frank Warmerdam <warmerdam@pobox.com>
+ * Copyright (c) 2010, Even Rouault <even dot rouault at mines-paris dot org>
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/dxf/ogrdxf_leader.cpp]: Not sure Frank or myself deserve copyright in this new file

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxf_leader.cpp [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "ogr_dxf.h"
+#include "cpl_conv.h"
+
+CPL_CVSID("$Id$")
+
+#define VECTOR_LEN(x,y) sqrt( (x)*(x) + (y)*(y) )
+#define POINT_DIST(x1,y1,x2,y2)  VECTOR_LEN((x2-x1),(y2-y1))
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/dxf/ogrdxf_leader.cpp]: perhaps turn those macros into inline functions ?

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxfdatasource.cpp [hunk 5]
```diff
@@ -508,6 +512,90 @@ const char *OGRDXFDataSource::LookupLineType( const char *pszName )
...
 
 /************************************************************************/
+/*                  PopulateDefaultDimStyleProperties()                 */
+/************************************************************************/
+
+void OGRDXFDataSource::PopulateDefaultDimStyleProperties(
+    std::map<CPLString, CPLString>& oDimStyleProperties)
+
+{
+    int* piCode = ACGetKnownDimStyleCodes();
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/dxf/ogrdxfdatasource.cpp]: const int*

**Comment by atlight** [comment]
Not sure where your review comments went, but I have addressed all of them.

**Comment by rouault** [comment]
Committed in r40569

**Pull request closed without merging**